File: Serene/Serene.Script/Imports/ServerImports/Northwind.ShipperForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class ShipperForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Shipper";

        public ShipperForm(string idPrefix) : base(idPrefix) {}

        public StringEditor CompanyName { [InlineCode("{this}.w('CompanyName', Serenity.StringEditor)")] get; private set; }
        public PhoneEditor Phone { [InlineCode("{this}.w('Phone', Serene.Northwind.PhoneEditor)")] get; private set; }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.ShipperForm.ts
namespace Serene.Northwind {
    export interface ShipperForm {
        CompanyName: Serenity.StringEditor;
        Phone: PhoneEditor;
    }

    export class ShipperForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Shipper';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!ShipperForm.init)  {
                ShipperForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;
                var w1 = PhoneEditor;

                Q.initFormType(ShipperForm, [
                    'CompanyName', w0,
                    'Phone', w1
                ]);
            }
        }
    }
}
